src/backend/vulkan/src/window.rs

```diff
@@ -34,69 +38,124 @@ impl Drop for RawSurface {
 }
 
 impl Instance {
-    pub fn create_surface(&self, window: &winit::Window) -> Surface {
+    #[cfg(unix)]
+    pub fn create_surface_from_xlib(&self, display: *mut c_void, window: usize) -> Surface {
         let entry = VK_ENTRY
```

Comment:
I see that we are no longer checking for extensions (e.g. `vk::VK_KHR_XLIB_SURFACE_EXTENSION_NAME`). I think we should, and not just verify that an extension is supported. `create_surface` that works with `winit` should pick a window creation method among the available ones based on the extension support (xlib/xcb/wayland/hwnd/etc).

Reply:
OK, I'll add extension checking back. One problem: should it panic once `create_surface_from_xlib` fails? If `vk:VK_KHR_XLIB_SURFACE_EXTENSION_NAME` is not supported, `create_surface_from_xlib` should return Err instead of panic IMO.